zproject/backends.py

```diff
@@ -1209,7 +1209,7 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
         # In SAML authentication, the IdP may support only sending
         # the first and last name as separate attributes - in that case
         # we construct the full name from them.
-        return_data["full_name"] = f"{first_name} {last_name}".strip()  # strip removes the unnecessary ' '
+        return_data["full_name"] = f"{first_name or ''} {last_name or ''}".strip()  # strip removes the unnecessary ' '
```

Comment:
Now that it's not just SAML using this code, we should update the comment above.

Reply:
Do these `or ''` do anything? They’re already looked up with a default of `''` above:

```python
    first_name = kwargs['details'].get('first_name', '')
    last_name = kwargs['details'].get('last_name', '')
```

Reply:
when the name details aren't sent by apple, python-social-auth [sets](github.com/python-social-auth/social-core/blob/master/social_core/backends/apple.py#L146#L150) `first_name` and `last_name` to `None`. Because of that return_data["full_name"] is getting the value `"None None"`. Those `or ''` are used to avoid that.

Reply:
In general this whole structure is kind of weird:
```
    if full_name is None:
        if not first_name and not last_name:
            # We need custom code here for any social auth backends
            # that don't provide name details feature.
            if (backend.name == 'apple'):
                # Apple authentication provides the user's name only
                # the very first time a user tries to login.  So if
                # the user aborts login or otherwise is doing this the
                # second time, we won't have any name data.  We handle
                # this by setting full_name to be the empty string.
                full_name = ""
            else:
                raise AssertionError("Social auth backend doesn't provide name")
    if full_name:
        return_data["full_name"] = full_name
    else:
        # In SAML authentication, the IdP may support only sending
        # the first and last name as separate attributes - in that case
        # we construct the full name from them.
        return_data["full_name"] = f"{first_name} {last_name}".strip()  # strip removes the unnecessary ' '
        return_data["full_name"] = f"{first_name or ''} {last_name or ''}".strip()  # strip removes the unnecessary ' '
```

The block `if (backend.name == 'apple'):` does nothing, because then we fall into the `else` condition of `if full_name / else`. I think this needs to be cleaned up.